pyro/contrib/tracking/assignment.py

```diff
@@ -89,10 +193,112 @@ def __init__(self, exists_logits, assign_logits, bp_iters=None):
         assert self.exists_dist.batch_shape == (self.num_objects,)
 
 
+def compute_marginals(exists_logits, assign_logits):
+    """
+    This implements exact inference of pairwise marginals via
+    enumeration. This is very expensive and is only useful for testing.
+
+    See :class:`MarginalAssignment` for args and problem description.
+    """
+    num_detections, num_objects = assign_logits.shape
+    assert exists_logits.shape == (num_objects,)
+
+    exists_probs = exists_logits.new_zeros(2, num_objects)  # [not exist, exist]
+    assign_probs = assign_logits.new_zeros(num_detections, num_objects + 1)
+    for assign in itertools.product(range(num_objects + 1), repeat=num_detections):
+        assign_part = sum(assign_logits[j, i] for j, i in enumerate(assign) if i < num_objects)
+        for exists in itertools.product(*[[1] if i in assign else [0, 1] for i in range(num_objects)]):
+            exists_part = sum(exists_logits[i] for i, e in enumerate(exists) if e)
+            prob = _exp(exists_part + assign_part)
+            for i, e in enumerate(exists):
+                exists_probs[e, i] += prob
+            for j, i in enumerate(assign):
+                assign_probs[j, i] += prob
+
+    # Convert from probs to logits.
+    exists = exists_probs.log()
+    assign = assign_probs.log()
+    exists = exists[1] - exists[0]
+    assign = assign[:, :-1] - assign[:, -1:]
+    _warn_if_nan(exists, 'exists')
+    _warn_if_nan(assign, 'assign')
+    return exists, assign
+
+
+def compute_marginals_bp(exists_logits, assign_logits, bp_iters):
```

Comment:
nit: i like `compute_marginals` for exact and `approximate_marginals_bp` for bp to evoke one is exact and another isnt